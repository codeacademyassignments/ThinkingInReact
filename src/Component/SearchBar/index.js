import React, {Component} from 'react';

const SearchBar = (props)=> { 
  return (
  <div>
  <input type="text" placeholder="Search..." value = {props.filterText} onChange = {(event)=>props.onTextChange(event.target.value)}/>
  <br></br>
  <input type="checkbox" checked = {props.inStock} onClick={(event)=>{props.onTickChange(event.target.checked)}} />Only show products in stock
  </div>);  
}
export default SearchBar;