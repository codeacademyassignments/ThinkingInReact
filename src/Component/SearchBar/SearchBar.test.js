import './index.js';
import renderer from 'react-test-renderer';
import SearchBar from './index';
import React from 'react';
import {shallow} from 'enzyme';

describe('searchBar',()=>{
  const onTickChangeMock = jest.fn();
  const onTextChangeMock = jest.fn();
  const wrapper = shallow(<SearchBar filterText='nothing' inStock={true} onTickChange = {onTickChangeMock} onTextChange = {onTextChangeMock}/>);
  //console.log(wrapper.props());
  it('snapshot should match',()=>{
    const tree = renderer.create(<SearchBar filterText='nothing' inStock={true} onTickChange = {onTickChangeMock} onTextChange = {onTextChangeMock}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
  it('should call onTextChangeMock when text changes',()=>{
 // expect(wrapper.props().filterText).toEqual('nothing');
//expect(wrapper.props.inStock).toEqual(true);
    wrapper.find('[type="text"]').simulate('change',{target:{value:'something'}});
   // expect(onTextChangeMock).toHaveBeenCalled();
    expect(onTextChangeMock).toHaveBeenCalledWith('something');
  })
  it('should call onTickChangeMock when tick checked/unchecked',()=>{
    //expect(wrapper.props().filterText).toEqual('nothing');
//expect(wrapper.props.inStock).toEqual(true);
    wrapper.find('[type="checkbox"]').simulate('click',{target:{checked:true}});
    expect(onTickChangeMock).toHaveBeenCalled();
    expect(onTickChangeMock).toHaveBeenCalledWith(true);
  })
})


