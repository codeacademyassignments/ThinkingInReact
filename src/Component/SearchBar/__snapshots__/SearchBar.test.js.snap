// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchBar snapshot should match 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value="nothing"
  />
  <br />
  <input
    checked={true}
    onClick={[Function]}
    type="checkbox"
  />
  Only show products in stock
</div>
`;
